pkg/orchestrator/azureDevOps.go

```diff
@@ -39,119 +90,97 @@ func (a *AzureDevOpsConfigProvider) OrchestratorType() string {
+		return []byte{}, nil
 	}
 	var responseInterface map[string]interface{}
 	err = piperHttp.ParseHTTPResponseBodyJSON(response, &responseInterface)
 	if err != nil {
-		log.Entry().Error("failed to parse http response", err)
-		return logs, nil
+		log.Entry().Error("failed to parse http response: ", err)
+		return []byte{}, err
 	}
 	// check if response interface is empty or non-existent
-	logCount := int(responseInterface["count"].(float64))
-
+	var logCount int
+	if val, ok := responseInterface["count"]; ok {
+		logCount = int(val.(float64))
+	} else {
+		log.Entry().Error("log count variable not found, returning empty log")
+		return []byte{}, err
+	}
+	var logs []byte
 	for i := 1; i <= logCount; i++ {
 		counter := strconv.Itoa(i)
 		logURL := URL + "/" + counter
-		fmt.Println("logURL: ", logURL)
 		log.Entry().Debugf("Getting log no.: %d  from %v", i, logURL)
 		response, err := a.client.GetRequest(logURL, nil, nil)
 		if err != nil {
-			fmt.Println(err)
+			log.Entry().Error("failed to get log", err)
+			return []byte{}, err
+		}
+		if response.StatusCode != 200 { //http.StatusNoContent -> also empty log!
+			log.Entry().Errorf("response code is %v, could not get log information from AzureDevOps ", response.StatusCode)
+			return []byte{}, err
 		}
 		content, err := ioutil.ReadAll(response.Body)
+		if err != nil {
+			log.Entry().Error("failed to parse http response", err)
+			return []byte{}, err
+		}
 		logs = append(logs, content...)
 	}
 
 	return logs, nil
 }
 
-// GetPipelineStartTime returns the pipeline start time
+// GetPipelineStartTime returns the pipeline start time in UTC
 func (a *AzureDevOpsConfigProvider) GetPipelineStartTime() time.Time {
-	// "2021-10-11 13:49:09+00:00"
-	timestamp := getEnv("SYSTEM_PIPELINESTARTTIME", "n/a")
-	replaced := strings.Replace(timestamp, " ", "T", 1)
-	parsed, err := time.Parse(time.RFC3339, replaced)
-	if err != nil {
-		log.Entry().Errorf("Could not parse timestamp. %v", err)
-		// Return 1970 in case parsing goes wrong
-		parsed = time.Date(1970, time.January, 01, 0, 0, 0, 0, time.UTC)
+	//"2022-03-18T07:30:31.1915758Z"
+	a.getAPIInformation()
+	if val, ok := a.apiInformation["startTime"]; ok {
+		parsed, err := time.Parse(time.RFC3339, val.(string))
+		if err != nil {
+			log.Entry().Errorf("could not parse timestamp, %v", err)
+			parsed = time.Time{}
+		}
+		return parsed.UTC()
 	}
-	return parsed
+	return time.Time{}.UTC()
 }
 
-func (a *AzureDevOpsConfigProvider) GetSystemCollectionURI() string {
-	return getEnv("SYSTEM_COLLECTIONURI", "n/a")
-}
-
-func (a *AzureDevOpsConfigProvider) GetTeamProjectId() string {
-	return getEnv("SYSTEM_TEAMPROJECTID", "n/a")
-}
-
-func (a *AzureDevOpsConfigProvider) GetBuildId() string {
-	return getEnv("BUILD_BUILDID", "n/a")
+// GetBuildID returns the BuildNumber displayed in the ADO UI
+func (a *AzureDevOpsConfigProvider) GetBuildID() string {
```

Comment:
Why didn't you rename it to GetBuildNumber if the build number is returned?

Reply:
Good point. It will be used in the internal lib for reporting and there wie use - orchestrator independent "GetBuildID"